lib/recurring_events.ex

```diff
@@ -136,8 +136,19 @@ defmodule RecurringEvents do
     |> drop_after(rules)
   end
 
-  defp drop_exclude(dates, %{exclude_date: excludes}) do
-    dates |> Stream.filter(&(&1 not in excludes))
+  defp drop_exclude(dates, %{exclude_date: [%Elixir.Date{} | _] = excludes}) do
+    dates
+    |> Stream.filter(&(&1 not in excludes))
+  end
+
+  defp drop_exclude(dates, %{exclude_date: [%NaiveDateTime{} | _] = excludes}) do
+    dates
+    |> Stream.filter(&(&1 not in excludes))
+  end
+
+  defp drop_exclude(dates, %{exclude_date: [date_range]}) do
```

Comment:
Is there a way to pattern match on `#DateRange`? This is a protocol, and it seems you can't match.